include/eve/detail/shuffle_v2/shuffle_v2_driver_fwd.hpp

```diff
@@ -36,17 +51,24 @@ template<typename NativeSelector> struct shuffle_v2_driver
   template<std::ptrdiff_t G, std::ptrdiff_t... I, simd_value T, typename... Ts>
   EVE_FORCEINLINE auto operator()(pattern_t<I...> p, eve::fixed<G> g, T x, Ts... xs) const noexcept
   {
-    static_assert((std::same_as<T, Ts> && ...), "You can only shuffle the same type");
```

Comment:
static assert had 2 issues:
1 - compilation does not halt. So, for example, there were insane errors with division by 0 from the depths of hell.
2 - this function is several layers deep